android/app/src/main/kotlin/com/yubico/authenticator/fido/FidoManager.kt

```diff
@@ -603,6 +606,47 @@ class FidoManager(
             ).toString()
         }
 
+    private suspend fun enableEnterpriseAttestation(): String =
+        connectionHelper.useSession(FidoActionDescription.EnableEnterpriseAttestation) { fidoSession ->
+            if (Config.isSupported(fidoSession.cachedInfo)) {
+                val uvAuthProtocol = getPreferredPinUvAuthProtocol(fidoSession.cachedInfo)
+                val clientPin = ClientPin(fidoSession, uvAuthProtocol)
+                val token = if (pinStore.hasPin()) {
+                    clientPin.getPinToken(
+                        pinStore.getPin(),
+                        ClientPin.PIN_PERMISSION_ACFG,
+                        null
+                    )
+                } else null
+                val config = Config(fidoSession, uvAuthProtocol, token)
+
+                try {
+                    config.enableEnterpriseAttestation()
+                    fidoViewModel.setSessionState(
+                        Session(
+                            fidoSession.info,
+                            pinStore.hasPin(),
+                            pinRetries
+                        )
+                    )
+                } catch (e: Exception) {
+                    logger.error("Failed to enable enterprise attestation. ", e)
+                    return@useSession JSONObject(
+                        mapOf(
+                            "success" to false,
+                        )
+                    ).toString()
+                }
+            } else {
+                logger.debug("authenticatorConfig not supported, ignoring call to enableEnterpriseAttestation")
+            }
+
+            return@useSession JSONObject(
+                mapOf(
+                    "success" to true,
+                )
+            ).toString()
```

Comment:
It seems strange to me that we would silently ignore and return `success: true` when the YubiKey doesn't support Entrprise Attestation. Shouldn't that result in a failure?

Reply:
sure! I will return a failure when this method is called and the authenticator does not support authenticatorConfig.

Reply:
updated